FirebaseFunctions/Sources/Functions.swift

```diff
@@ -462,33 +516,27 @@ enum FunctionsConstants {
       fetcher.allowedInsecureSchemes = ["http"]
     }
 
-    fetcher.beginFetch { [self] data, error in
-      let result: Result<HTTPSCallableResult, any Error>
-      do {
-        let data = try responseData(data: data, error: error)
-        let json = try responseDataJSON(from: data)
-        // TODO: Refactor `decode(_:)` so it either returns a non-optional object or throws
-        let payload = try serializer.decode(json)
-        // TODO: Remove `as Any` once `decode(_:)` is refactored
-        result = .success(HTTPSCallableResult(data: payload as Any))
-      } catch {
-        result = .failure(error)
-      }
+    return fetcher
+  }
 
-      DispatchQueue.main.async {
-        completion(result)
-      }
-    }
+  private func callableResultFromResponse(data: Data?,
+                                          error: (any Error)?) throws -> HTTPSCallableResult {
+    let processedData = try processedResponseData(from: data, error: error)
+    let json = try responseDataJSON(from: processedData)
+    // TODO: Refactor `decode(_:)` so it either returns a non-optional object or throws
+    let payload = try serializer.decode(json)
+    // TODO: Remove `as Any` once `decode(_:)` is refactored
+    return HTTPSCallableResult(data: payload as Any)
   }
 
-  private func responseData(data: Data?, error: (any Error)?) throws -> Data {
+  private func processedResponseData(from data: Data?, error: (any Error)?) throws -> Data {
     // Case 1: `error` is not `nil` -> always throws
     if let error = error as NSError? {
       let localError: (any Error)?
       if error.domain == kGTMSessionFetcherStatusDomain {
         localError = FunctionsError(
           httpStatusCode: error.code,
-          body: data,
+          body: data ?? error.userInfo["data"] as? Data,
```

Comment:
The failing integration tests have been fixed with this change.

<details>
<summary>Rationale</summary>

* `GTMSessionFetcher`’s async `beginFetch()` method returns either data or error, but not both (unlike the completion-handler variant)
* Without the data, `processedResponseData(from:error:)` has less details and can’t categorize the error correctly
* However, the same details can be retrieved from the error’s user info
</details>